datamade/openness-project-nmid
# Convert available funds / contributions/expenditures from area to step charts

## Overview

This PR updates the charts on candidate and committee pages. Namely it:

- Changes them from area charts to step charts, see https://github.com/datamade/openness-project-nmid/pull/216#issuecomment-2386614376
- Updates chart titles, adds axis labels
- Forces a consistent date range along the x-axis for easier comparison, see https://github.com/datamade/openness-project-nmid/pull/216#issuecomment-2389141748
- Fixes tooltip formatting
- Updates the available funds chart to use the end of the filing period, rather than the date the filing was made

It also updates the aggregate queries to force a continuous interval of aggregate data. Previously, it was sparse, which posed a challenge for charting. See https://github.com/datamade/openness-project-nmid/pull/216#discussion_r1786566219

### Demo

![charts](https://github.com/user-attachments/assets/3f4ff7b7-6298-4347-8c00-85b568b51422)

## Testing instructions

Update your local Docker Compose file to connect to a read-only version of the prod database (see Money Trail NM - Docker Env - Readonly Prod DB note in Bitwarden), then view some donation / expenditure charts and confirm they look good.
